### ISSUE TITLE
bug: MediaQueryList.addEventListener('change') stopped working in tests in 3.2.1

### ISSUE THREAD

**Issue by asik** [ionic-team/stencil#4489]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

3.2.1

### Current Behavior

Repro is simple, just `yarn create stencil` and add this code to `my-component.tsx`:
```ts
  #mediaQueryList: MediaQueryList = null;

  componentWillRender() {
    this.#mediaQueryList = window.matchMedia(`(max-width: 4px)`);
    this.#mediaQueryList.addEventListener('change', () => {});
  }
```

Then run `yarn test`.

You get this error:

```
 FAIL  src/components/my-component/my-component.spec.ts
  ● my-component › renders

    TypeError: Cannot create property '__listeners' on string 'change'

      31 |   componentWillRender() {
      32 |     this.#mediaQueryList = window.matchMedia(`(max-width: 4px)`);
    > 33 |     this.#mediaQueryList.addEventListener('change', () => {});
         |                          ^
      34 |   }
      35 |
      36 |   render() {

      at Object.addEventListener (node_modules/@stencil/core/mock-doc/index.cjs:685:24)
      at MyComponent.componentWillRender [as __componentWillRender] (src/components/my-component/my-component.tsx:33:26)
      at MyComponent.e.componentWillRender (node_modules/@stencil/core/testing/index.js:4255:20)
      at safeCall (node_modules/@stencil/core/internal/testing/index.js:491:14)
      at node_modules/@stencil/core/internal/testing/index.js:428:56
      at enqueue (node_modules/@stencil/core/internal/testing/index.js:430:52)
      at dispatchHooks (node_modules/@stencil/core/internal/testing/index.js:428:38)
      at a (node_modules/@stencil/core/internal/testing/index.js:419:18)
      at a (node_modules/@stencil/core/internal/testing/index.js:30:17)
```

This started happening in 3.2.1. 
Someone noticed this before: https://github.com/ionic-team/stencil/issues/4193#issuecomment-1521019769
A workaround is to use the `onchange` function instead.

### Expected Behavior

The tests should pass

### Steps to Reproduce

See above

### Code Reproduction URL

.

**Comment by rwaskiewicz** [comment]
Thanks @asik!

I was able to confirm that this is an issue for `MediaQueryList` in Mock Doc - I've labeled this to get it into the team's backlog.

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.0.2 release](https://github.com/ionic-team/stencil/releases/tag/v4.0.2)
